src/rgw/rgw_rest_s3.cc

```diff
@@ -3229,6 +3229,8 @@ int RGW_Auth_S3::authorize(RGWRados *store, struct req_state *s)
     return 0;
   }
 
+  s->aws4_auth_presigned_url_bugfix = false;
+
   if (!s->http_auth || !(*s->http_auth)) {
```

Comment:
The ' else' part also has a call to authorize_v4, I think the same logic of computing the signature twice, needs to be added there also.